thirdeye-ui/src/app/components/alert-composite-filters-modal/alert-composite-filters-modal.component.tsx

```diff
@@ -0,0 +1,231 @@
+import { CohortTableRowData } from "../cohort-detector/cohorts-table/cohorts-table.interfaces";
+import {
+    getCohortTableRowFromData,
+    NAME_JOIN_KEY,
+} from "../cohort-detector/cohorts-table/cohorts-table.utils";
+import { DatasetDetails } from "../cohort-detector/dataset-details/dataset-details.component";
+import { Modal } from "../modal/modal.component";
+import { AlertCompositeFiltersModalProps } from "./alert-composite-filters-modal.interfaces";
+
+export const AlertCompositeFiltersModal: FunctionComponent<AlertCompositeFiltersModalProps> =
+    ({ onUpdateCompositeFiltersChange, onCancel }) => {
+        const { t } = useTranslation();
+        const { notify } = useNotificationProviderV1();
+
+        const { cohortsResponse, getCohorts, status, errorMessages } =
+            useGetCohort();
+
+        const { alert, onAlertPropertyChange, setIsMultiDimensionAlert } =
+            useOutletContext<AlertCreatedGuidedPageOutletContext>();
+
+        const [selectedCohorts, setSelectedCohorts] = useState<CohortResult[]>(
+            []
+        );
+
+        const [queryValue, setQueryValue] = useSessionStorage<string>(
+            SessionStorageKeys.QueryFilterOnAlertFlow,
+            ""
+        );
+
+        useEffect(() => {
+            setIsMultiDimensionAlert(true);
+        }, []);
+
+        useEffect(() => {
+            notifyIfErrors(
+                status,
+                errorMessages,
+                notify,
+                t("message.error-while-fetching", {
+                    entity: t("label.dimensions-data"),
+                })
+            );
+        }, [status]);
+
+        const handleSearchButtonClick = (
+            getCohortsParams: GetCohortParams
+        ): void => {
+            const params = { ...getCohortsParams };
+
+            if (params.aggregationFunction === MetricAggFunction.COUNT) {
+                params.roundOffThreshold = true;
+            }
+
+            getCohorts(params);
+        };
+
+        const handleCohortsSelectionChange = (
+            cohorts: CohortResult[]
+        ): void => {
+            setSelectedCohorts(cohorts);
+        };
+
+        const initiallySelectedCohorts = useMemo<CohortTableRowData[]>(() => {
+            if (!(status === ActionStatus.Done && cohortsResponse)) {
+                return [];
+            }
+
+            // Extract all the queryFilters in alert
+            const selectedEnumerationItemsInAlert = (
+                (alert.templateProperties.enumerationItems ||
+                    []) as EnumerationItemConfig[]
+            ).map((item) => item.params.queryFilters);
+
+            // Parse the stringified `queryFilter=` to an object
+            // `dimensionFilters` format for precise comparison
+            const selectedDimensionFilters = selectedEnumerationItemsInAlert
+                .map((query) =>
+                    query
+                        .split(NAME_JOIN_KEY)
```

Comment:
Just curious, has this code snipped been copied from somewhere else in the codebase?

Reply:
@alcatraz627 The same logic is being used in src/app/pages/alerts-create-guided-page/setup-dimension-groups/setup-dimension-groups-page.component.tsx file line 117.